src/seedu/addressbook/commands/FindCommand.java

```diff
@@ -50,11 +51,40 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            if (!IsDisjointSet(wordsInName, keywords)) {
                 matchedPersons.add(person);
             }
         }
         return matchedPersons;
     }
 
+    /**
+     * Check if both set are disjoint
+     *
+     * @param firstSet
+     * @param secondSet
+     * @return true if two both set are disjoint
+     */
+    private boolean IsDisjointSet(Set<String> firstSet, Set<String> secondSet) {
+        return Collections.disjoint(toLower(firstSet), toLower(secondSet));
+    }
+
```

Comment:
I see that you are converting `Set<String>` to array, working with the array and then formatting the output back into `Set<String>` again.

There is a way to simplify this. Using an enhance for loop to iterate through the `String` variables in `Set<String> strings`, you can just do `returnString.add(xxx.toLowerCase())` while iterating through `Set<String> strings`. That in my opinion would be much neater :)

Reply:
oh yea! i can do that. Thanks for pointing it out